src/libraries/Microsoft.PowerFx.Core/Public/CheckResult.cs

```diff
@@ -62,6 +68,46 @@ public void ThrowOnErrors()
             }
         }
 
+        /// <summary>
+        /// Gets the type of a syntax node.
+        /// </summary>
+        /// <param name="node"></param>
+        /// <returns></returns>
+        public FormulaType GetNodeType(TexlNode node) => FormulaType.Build(_binding.GetType(node));
+
+        /// <summary>
+        /// Whether the node is a constant node.
+        /// </summary>
+        /// <param name="node"></param>
+        /// <returns></returns>
+        public bool IsNodeConstant(TexlNode node) => _binding.IsConstant(node);
+
+        /// <summary>
+        /// Check whether function invocation in type-correct.
+        /// </summary>
+        /// <param name="fnc"></param>
+        /// <param name="nodes"></param>
+        /// <param name="types"></param>
+        /// <param name="retType"></param>
+        /// <returns></returns>
+        public bool CheckInvocation(FunctionInfo fnc, IReadOnlyList<TexlNode> nodes, IReadOnlyList<FormulaType> types, out FormulaType retType)
```

Comment:
Why do we want to expose CheckInvocation?

Reply:
Ideally, we'd really like to move away from using CheckInvocation as our method of validating params. Exposing it (and it's oddities) publicly seems like the wrong direction.

Reply:
@ivanradicek - can we just pass the candidates as strings into a Check() call?

Reply:
This is quite important for our use cases. String is problematic because we don't have mapping from names to functions at this point. What can we do about this?